crates/sui-framework/packages/sui-system/sources/sui_system.move

```diff
@@ -596,6 +596,13 @@ module sui_system::sui_system {
         inner
     }
 
+    // For read only operations
+    fun load_inner(self: &SuiSystemState): &SuiSystemStateInnerV2 {
```

Comment:
How do you handle sui system state version upgrades? The reason this is never a read-only operation is because of the need to support lazy upgrade

Reply:
Does the version upgrade happens during end of epoch? Can we assume that by the time a user calls the read only function in the new epoch, the version will have already been updated?

Reply:
don't know why I thought system state can't be use mutably, please ignore :)